pkg/upstream/cluster/cluster_manager.go

```diff
@@ -79,9 +80,19 @@ func (singleton *clusterManagerSingleton) Destroy() {
 	clusterManagerInstance.clusterManager = nil
 }
 
+type ClusterManagerOptions func(types.ClusterManager)
+
+func WithClusterPoolEnable(clusterPoolEnable bool) ClusterManagerOptions {
+	return func(cm types.ClusterManager) {
+		if c, ok := cm.(*clusterManagerSingleton); ok {
+			c.clusterPoolEnable = clusterPoolEnable
+		}
+	}
+}
+
 var clusterManagerInstance = &clusterManagerSingleton{}
 
-func NewClusterManagerSingleton(clusters []v2.Cluster, clusterMap map[string][]v2.Host, tls *v2.TLSConfig) types.ClusterManager {
+func NewClusterManagerSingleton(clusters []v2.Cluster, clusterMap map[string][]v2.Host, tls *v2.TLSConfig, initFns ...ClusterManagerOptions) types.ClusterManager {
```

Comment:
`Option` but not `Options`, and parameter should be renamed to `initOptions` or just `opts`

Reply:
🆗